pennylane_cirq/cirq_device.py

```diff
@@ -170,6 +182,17 @@ def __init__(self, wires, shots, qubits=None):
         "Projector": CirqOperation(lambda: cirq.ProductState.projector),
     }
 
+    def supports_operation(self, operation):
+        # pylint: disable=missing-function-docstring
+        op = operation
+        if isinstance(op, type) and issubclass(op, Operation):
+            op = op.name
+        if isinstance(op, str):
+            op_with_power = op.split("**")
+            if len(op_with_power) == 2 and "Pow_" + op_with_power[0] in self._pow_operation_map:
+                return True
```

Comment:
To avoid creating more branches, we could just do sth like:
```suggestion
        if isinstance(op, str) and len(op.split("**")) == 2:
            operation = "Pow_" + op_with_power[0]
```
And then the check will be done in `super().supports_operation`

Reply:
Taking a deeper look at the code, it is weird that `self._complete_operation_map` is only used in `self._apply_operation` and not used in `self.operations` (which is the list used to see if the operation is supported or not).

Shouldn't we do:
```python
@property
def operations(self):
    # pylint: disable=missing-function-docstring
    return set(self._complete_operation_map.keys())
```

Reply:
Your refactor makes me split the string twice which I don't love, but I'll think of a better way.

I didn't update to use `_complete_operation_map` because the `supports_operation` method is typically invoked with `obj.name`, so something like `qml.PauliX(0)**1.1` will ultimately be validated with `supports_operation("PauliX**1.1")` instead of `"Pow(PauliX)"` (which would match the `Adjoint(S)` pattern used above). Unless we changed the `qml.ops.Pow.name` property to something like `Pow(PauliX)`, we need to do something "hacky" on the plugin side to support powers. And I don't think we should change the name.

Reply:
I didn't mean to remove the `supports_operation` method overriding, I like this. But by using `_complete_operation_map` in `self.operations` you could change the operation name to `"Pow(PauliX)"` and then send it to `super().supports_operation`.

Something like:
```python
def supports_operation(self, operation):
    # pylint: disable=missing-function-docstring
    if isinstance(operation, str):
        op_with_power = operation.split("**")
        operation = "Pow_" + op_with_power[0] if len(op_with_power) == 2 else operation
    return super().supports_operation(operation)
"""